reservoirpy/datasets/_chaos.py

```diff
@@ -268,20 +270,28 @@ def mackey_glass(
             on Wikipedia.
 
     """
-    # a random state is needed as the method used to discretize
-    # the timeseries needs to use randomly generated initial steps
-    # based on the initial condition passed as parameter.
-    if seed is None:
-        seed = get_seed()
-
-    rs = rand_generator(seed)
+    
+    if history is None:
+	    # a random state is needed as the method used to discretize
+	    # the timeseries needs to use randomly generated initial steps
+	    # based on the initial condition passed as parameter.
+        if seed is None:
+            seed = get_seed()
+
+            rs = rand_generator(seed)
+
+            # generate random first step based on the value
+            # of the initial condition
+            history_length = int(np.floor(tau / h))
+            history = collections.deque(
+            x0 * np.ones(history_length) + 0.2 * (rs.random(history_length) - 0.5)
+            )
+    else:
+        assert int(round(tau / h)) <= len(history), \
```

Comment:
I think this should be a ValueError. AssertionError are not specific enough. And maybe the error text is a bit puzzling: maybe you could just state that history must have a length of a specific value, and add in parenthesis that this value is equal to tau/h ?